Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -132,6 +137,15 @@
             "helpMarkDown": "Additional Web Deploy arguments that will be applied when deploying the Azure Web App like,-disableLink:AppPoolExtension -disableLink:ContentExtension."
         },
         {
+            "name": "VariableSubstitution",
+            "type": "boolean",
+            "label": "Variable Substitution",
+            "required": false,
+            "defaultValue": "false",
+            "groupName": "configtransforms",
+            "helpMarkDown": "Variables defined in the Defination will be automatically substitued in the appSettings,applicationSettings and connectionStrings section of the config files. If the same variable are defined in the Defination and Environment, then the En variables will be supersede the defination variables. Variable substitution is run after config Transformations."
```

Comment:
We will have to add that secure variable support is not available for substitution.